inst/include/dust2/r/continuous/filter.hpp

```diff
@@ -0,0 +1,95 @@
+#pragma once
+
+#include <monty/r/random.hpp>
+#include <dust2/r/helpers.hpp>
+#include <dust2/r/filter.hpp>
+#include <dust2/r/continuous/control.hpp>
+#include <dust2/filter.hpp>
+#include <dust2/continuous/system.hpp>
+
+namespace dust2 {
+namespace r {
+
+template <typename T>
+cpp11::sexp dust2_continuous_filter_alloc(cpp11::list r_pars,
+                                          cpp11::sexp r_time_start,
+                                          cpp11::sexp r_time,
+                                          cpp11::list r_ode_control,
+                                          cpp11::list r_data,
+                                          cpp11::sexp r_n_particles,
+                                          cpp11::sexp r_n_groups,
+                                          cpp11::sexp r_n_threads,
+                                          cpp11::sexp r_index_state,
+                                          cpp11::sexp r_seed) {
+  using rng_state_type = typename T::rng_state_type;
+  using rng_seed_type = std::vector<typename rng_state_type::int_type>;
+  using real_type = typename  T::real_type;
+
+  const auto n_particles = to_size(r_n_particles, "n_particles");
+  const auto n_groups = to_size(r_n_groups, "n_groups");
+  const auto n_threads = to_size(r_n_threads, "n_threads");
+  const auto time_start = check_time(r_time_start, "time_start");
+  const auto time = check_time_sequence(time_start, r_time, true, "time");
+  const auto ode_control = validate_ode_control<real_type>(r_ode_control);
+  const auto shared = build_shared<T>(r_pars, n_groups);
+  const auto internal = build_internal<T>(shared, n_threads);
+  const auto data = check_data<T>(r_data, shared, time.size(), "data");
+
+  // It's possible that we don't want to always really be
+  // deterministic here?  Though nooone can think of a case where
+  // that's actually the behaviour wanted.  For now let's go fully
+  // deterministic.
```

Comment:
I think this comment only applies to the unfilter?

Reply:
yeah, you're right